### ISSUE TITLE
Bug in Mean calculation in formatter?

### ISSUE THREAD

**Issue by mscuthbert** [vexflow/vexflow#187]
There seems to be a bug in Formatter.tune()'s embedded function "updateStats".

It is called for each Tickable with its duration (converted to a string of a reduced Fraction, like '1/2') and the amount of space off from the ideal for that duration it has:

```typescript
    function updateStats(duration: string, space: number) {
      const stats = durationStats[duration];
      if (stats === undefined) {
        durationStats[duration] = { mean: space, count: 1 };
      } else {
        stats.count += 1;
        stats.mean = (stats.mean + space) / 2;
      }
    }
```

The Mean seems to be calculated incorrectly.  If there are 3 quarter notes with space-differences of -2, 0, 2, then their mean deviation is obviously 0.  But here it would do:

```
-2  => mean = -2
0  => mean = -1
2 => mean = 0.5
```

I have a patch for this along with some other light refactors (putting the embedded function "move" on TickContext rather than deeply nested inside Formatter, for instance) I'm happy to PR if this sounds like a solution.  Just wanted to make sure that this wasn't intentional.  Note though that this correction is making no difference right now in the generated images (except for the annotations at the bottom) so I'm not entirely sure why yet.

There's a second point I'm not sure is intentional or not:  TickContexts with more Tickables potentially get a lot more weight in the tuning process (because the deviation cost is calculated as a sumArray, not a sumArray / numTickables):

```typescript
const cost = -sumArray(context.getTickables().map((t) => t.getFormatterMetrics().space.deviation));
```
rather than

```typescript
const cost = -sumArray(context.getTickables().map((t) => t.getFormatterMetrics().space.deviation)) 
    / context.getTickables().length;  // with div by 0 checks...
```

Also happy to include that fix, but maybe we care more about proper positioning the more tickables there are in a context?  (seems odd though).

**Comment by rvilarl** [comment]
Please put the PR and we will review.

**Comment by sschmidTU** [comment]
What's interesting is that I don't get any visual changes (regressions) over 211 samples when I do the osmd visual regression tests when applying the updateStats() fix (in #188). So it looks like this didn't affect the most used systems much.
Still good to fix the bug of course.
